ts/components/services/ContactPreferencesToggles/index.tsx

```diff
@@ -1,91 +1,178 @@
+import { serviceIDCurrentSelector } from "../../../store/reducers/entities/services/currentService";
+import {
+  isServicePreferenceResponseSuccess,
+  ServicePreference
+} from "../../../types/services/ServicePreferenceResponse";
+import { isStrictSome } from "../../../utils/pot";
+import { showToast } from "../../../utils/showToast";
 import PreferenceToggleRow from "./PreferenceToggleRow";
 
 type Item = "email" | "push" | "inbox";
 
 type Props = {
   channels?: ReadonlyArray<NotificationChannelEnum>;
-} & ReturnType<typeof mapStateToProps>;
+} & ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps>;
+
+const hasChannel = (
+  channel: NotificationChannelEnum,
+  channels?: ReadonlyArray<NotificationChannelEnum>
+) =>
+  fromNullable(channels)
+    .map(anc => anc.indexOf(channel) !== -1)
+    .getOrElse(true);
+
+/**
+ * Utility function to get the user preference value for a specific channel
+ * return false if preference state is pot.none or if an error occurred on API Response
+ * */
+const getChannelPreference = (
+  potServicePreference: ServicePreferenceState,
+  key: Item
+): boolean => {
+  if (
+    isStrictSome(potServicePreference) &&
+    isServicePreferenceResponseSuccess(potServicePreference.value)
+  ) {
+    return potServicePreference.value.value[key];
+  }
+  return false;
+};
 
 const ContactPreferencesToggle: React.FC<Props> = (props: Props) => {
-  // functions from actions
-  const onValueChange: (item: Item, value: boolean) => void = (_item, _value) =>
-    undefined;
-
-  /*
-   *  TODO State should be removed when data will be available from store
-   */
-  const [inboxSwitched, setInboxSwitched] = useState(true);
-  const [pushSwitched, setPushSwitched] = useState(true);
-  const [emailSwitched, setEmailSwitched] = useState(true);
-
-  const onInboxValueChange = (value: boolean) => {
-    onValueChange("inbox", value);
-    setInboxSwitched(value);
+  const { isLoading, isError } = props;
+  const [isFirstRender, setIsFirstRender] = useState(true);
+
+  const loadPreferences = () => {
+    if (props.currentService !== null) {
+      props.loadServicePreference(props.currentService.serviceID);
+    }
   };
 
-  const hasChannel = (channel: NotificationChannelEnum) =>
-    fromNullable(props.channels)
-      .map(anc => anc.indexOf(channel) !== -1)
-      .getOrElse(true);
+  useEffect(loadPreferences, []);
 
-  const onPushValueChange = (value: boolean) => {
-    onValueChange("push", value);
-    setPushSwitched(value);
-  };
+  useEffect(() => {
+    if (!isFirstRender) {
+      if (isError) {
+        showToast(I18n.t("global.genericError"));
+      }
+    } else {
+      setIsFirstRender(false);
+    }
+  }, [isError]);
 
-  const onEmailValueChange = (value: boolean) => {
-    onValueChange("email", value);
-    setEmailSwitched(value);
+  const onValueChange = (value: boolean, type: Item) => {
+    if (
+      isStrictSome(props.servicePreferenceStatus) &&
```

Comment:
```suggestion
      pot.isSome(potServicePreference) &&
```
`isStrictSome` forces to considering only when it is some (not some updating or loading)
in this way the user experiences the push notification toggle disappear 

Could the suggestion solve?

### with 2s of network delay
https://user-images.githubusercontent.com/822471/124565619-87a6de80-de42-11eb-8fb8-e96155281275.mov